x-pack/plugin/eql/qa/common/src/main/java/org/elasticsearch/test/eql/EqlSpecLoader.java

```diff
@@ -55,6 +55,14 @@ private static String getTrimmedString(TomlTable table, String key) {
             spec.note(getTrimmedString(table, "note"));
             spec.description(getTrimmedString(table, "description"));
 
+            // default these values to false
+            spec.caseSensitiveOnly(Boolean.TRUE.equals(table.getBoolean("case_sensitive")));
+            spec.caseInsensitiveOnly(Boolean.TRUE.equals(table.getBoolean("case_insensitive")));
+
+            if (spec.supportsCaseInsensitive() == false && spec.supportsCaseSensitive() == false) {
+                throw new IllegalArgumentException("Invalid case (in)sensitivity for test: " + spec.toString());
```

Comment:
Since we know what is the issue, maybe be more descriptive in the error message, maybe something around "a test cannot be case sensitive only and case insensitive only at the same time".

Reply:
addressed in 254b5c3